src/promptflow/promptflow/batch/_base_executor_proxy.py

```diff
@@ -88,26 +96,31 @@ async def exec_aggregation_async(
         run_id: Optional[str] = None,
     ) -> AggregationResult:
         # ensure service health
-        await self.ensure_executor_health()
+        await self._ensure_executor_health()
         # call aggregation api to get aggregation result
         async with httpx.AsyncClient() as client:
             url = self.api_endpoint + "/Aggregation"
             payload = {"run_id": run_id, "batch_inputs": batch_inputs, "aggregation_inputs": aggregation_inputs}
             response = await client.post(url, json=payload, timeout=LINE_TIMEOUT_SEC)
-        response = self.process_http_response(response)
-        return AggregationResult.deserialize(response)
+        result = self._process_http_response(response)
+        return AggregationResult.deserialize(result)
 
-    def process_http_response(self, response: httpx.Response):
-        status_code = response.status_code
-        if status_code == 200:
+    def _process_http_response(self, response: httpx.Response):
```

Comment:
Maybe add comment here? The return value of this method can either be "response.json()" or "error", depending on the status code.